libfreerdp/core/settings.c

```diff
@@ -471,7 +472,37 @@ rdpSettings* freerdp_settings_new(DWORD flags)
 		settings->HomePath = GetKnownPath(KNOWN_PATH_HOME);
 		if (!settings->HomePath)
 				goto out_fail;
-		settings->ConfigPath = GetKnownSubPath(KNOWN_PATH_XDG_CONFIG_HOME, "freerdp");
+
+		/* For default FreeRDP continue using same config directory
+		 * as in old releases.
+		 * Custom builds use <Vendor>/<Product> as config folder. */
+		if (_stricmp(FREERDP_VENDOR_STRING, FREERDP_PRODUCT_STRING))
+		{
+			base = GetKnownSubPath(KNOWN_PATH_XDG_CONFIG_HOME,
+					       FREERDP_VENDOR_STRING);
+			if (base)
```

Comment:
@hardening Already handled. Next line after the end of the scope is a check for ```settings->ConfigPath == NULL``` which it is as long as it is not set.